src/user/formats/hal.ts

```diff
@@ -84,17 +84,80 @@ export function item(user: Principal, privileges: PrivilegeMap, hasControl: bool
   }
   if (isAdmin) {
     hal._links['password'] = {
-      href: '/user/' + user.id + '/password',
+      href: `/user/${user.id}/password`,
       title: 'Change user\'s password',
       hints: {
         allow: ['PUT'],
       }
     };
+    hal._links['edit-form'] = {
+      href: `/user/${user.id}/edit`,
+      title: `Edit ${user.nickname}`
+    };
   }
   return hal;
 
 }
 
+export function edit(user: Principal): HalResource {
+  return {
+    _links: {
+      self: {
+        href: `/user/${user.id}`,
+      },
+      up: {
+        href: `/user/${user.id}`,
+        title: 'Cancel',
+      },
+    },
+    _templates: {
+      default: {
+        title: 'Edit User',
+        method: 'POST',
+        contentType: 'application/x-www-form-urlencoded',
+        properties: [
+          {
+            name: 'identity',
+            prompt: 'Identity',
+            type: 'text',
+            value: user.identity,
+          },
+          {
+            name: 'nickname',
+            prompt: 'Nickname',
+            type: 'text',
+            value: user.nickname,
+          },
+
+          {
+            name: 'active',
+            prompt: 'Active',
+            options: {
+              inline: [
+                { prompt: 'Activated', value: 'true' },
+                { prompt: 'Deactivated', value: 'false' },
+              ],
+            },
+            value: user.active ? 'true' : 'false',
+          },
+          {
+            name: 'type',
+            prompt: 'Type',
+            options: {
+              inline: [
+                { prompt: 'User', value: 'user' },
+                { prompt: 'App', value: 'app' },
+                { prompt: 'Group', value: 'group' },
+              ],
+            },
+            value: user.type,
```

Comment:
It shouldn't be possible to change the type, because this can have all kinds of weird effects.